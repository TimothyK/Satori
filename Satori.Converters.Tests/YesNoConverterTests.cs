using System.Text.Json;
using System.Text.Json.Serialization;
using Shouldly;
using Snapshooter.MSTest;

namespace Satori.Converters.Tests;

[TestClass]
public class YesNoConverterTests
{
    #region Act

    private static string Serialize(Survey survey)
    {
        return JsonSerializer.Serialize(survey, new JsonSerializerOptions() {WriteIndented = true});
    }

    #endregion Act

    [TestMethod]
    public void Write()
    {
        //Arrange
        var survey = new Survey
        {
            Question1Response = true,
            Question2Response = false,
            Question3Response = true,
        };

        //Act
        var json = Serialize(survey);

        //Assert
        Snapshot.Match(json);
    }
    
    [TestMethod]
    public void Read()
    {
        //Arrange
        const string payload = 
            """
            {
                "Question1Response": "Yes",
                "Question2Response": "No"
            }
            """;

        //Act
        var survey = JsonSerializer.Deserialize<Survey>(payload);

        //Assert
        survey.ShouldNotBeNull();
        survey.Question1Response.ShouldBeTrue();
        survey.Question2Response.ShouldBeFalse();
    }
}

internal class Survey
{
    [JsonConverter(typeof(YesNoConverter))]
    public bool Question1Response { get; set; }
    [JsonConverter(typeof(YesNoConverter))]
    public bool Question2Response { get; set; }
    public bool Question3Response { get; set; }
}