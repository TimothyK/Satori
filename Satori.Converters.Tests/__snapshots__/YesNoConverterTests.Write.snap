{
  "Question1Response": "Yes",
  "Question2Response": "No",
  "Question3Response": true
}
